src/System.Diagnostics.Tracing.Telemetry/src/System/Diagnostics/Tracing/TelemetryListener.cs

```diff
@@ -0,0 +1,249 @@
+    /// 'quick checks'.   
+    /// 
+    /// The item in the stream is a KeyValuePair<string, object> where the string is the name
+    /// of the telemetry item and the object is the payload (typically an anonumous type).  
+    /// 
+    /// There may be many TelemetryListeners in the system, but we encourage the use of
+    /// The TelemetrySource.DefaultSource which goes to the TelemetryListener.DefaultListener.
+    /// 
+    /// If you need to see 'everything' you can subscribe to the 'AllListeners' event that
+    /// will fire for every live TelemetryListener in the appdomain (past or present). 
+    /// </summary>
+    public class TelemetryListener : TelemetrySource, IObservable<KeyValuePair<string, object>>, IDisposable
+    {
+        /// <summary>
+        /// When a TelemetryListener is created it is given a name.   Return this.  
+        /// </summary>
+        public string Name { get; private set; }
+        /// <summary>
+        /// This is the TelemetryListener that is used by default by the class library.   
+        /// Generally you don't want to make your own but rather have everyone use this one, which
+        /// insures that everyone who wished to subscribe gets the callbacks.  
+        /// The main reason not to us this one is that you WANT isolation from other 
+        /// events in the system (e.g. multi-tenancy).  
+        /// </summary>
+        public static TelemetryListener DefaultListener { get { return s_default; } }
+
+        /// <summary>
+        /// When you subscribe to this you get callbacks for all NotificationListeners in the appdomain
+        /// as well as those that occured in the past, and all future Listeners created in the future. 
+        /// </summary>
+        public static event Action<TelemetryListener> AllListeners
+        {
+            add
+            {
+                lock (DefaultListener)
+                {
+                    s_allListenersCallback = (Action<TelemetryListener>)Delegate.Combine(s_allListenersCallback, value);
+
+                    // Call back for each existing listener on the new callback.  
+                    for (TelemetryListener cur = s_allListeners; cur != null; cur = cur.m_next)
+                        value(cur);
+                }
+            }
+            remove
+            {
+                s_allListenersCallback = (Action<TelemetryListener>)Delegate.Remove(s_allListenersCallback, value);
+            }
+        }
+
+        // Subscription implementation 
+        /// <summary>
+        /// Add a subscriber (Observer).  If 'IsEnabled' == null (or not present), then the Source's IsEnabled 
+        /// will always return true.  
+        /// </summary>
+        virtual public IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Predicate<string> isEnabled)
+        {
+            // If we have been disposed, we silently ingore any subscriptions.  
+            if (m_disposed)
```

Comment:
This has been fixed (I ran the code through a styling tool that in fact fix this.